docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+# Installing OpenShift on OpenStack User-Provisioned Infrastructure
+
+The User-Provisioned Infrastructure (UPI) process installs OpenShift in stages, providing opportunities for modifications or integrating with existing infrastructure.
+
+It contrasts with the fully-automated Installer-Provisioned Infrastructure (IPI) which creates everything in one go.
+
+With UPI creating the cloud (OpenStack) resources (e.g. Nova servers, Neutron ports, security groups) is the responsibility of the person deploying OpenShift.
+
+The installer is still used to generate the ignition files and monitor the installation process.
+
+This provides a greater flexibility at the cost of a more explicit and interactive process.
+
+
+## Prerequisites
+
+The requirements for UPI are broadly similar to the [ones for OpenStack IPI][ipi-reqs]:
+
+[ipi-reqs]: ./openstack#openstack-requirements
+
+- OpenStack account with `clouds.yaml`
+  - This document will use a cloud called `openstack`
+- Nova flavors
+  - This document will use `m1.xlarge` for masters and `m1.large` for workers
+- An external subnet you want to use for floating IP addresses
+  - This document will use `external`
+- The `openshift-install` binary in your `$PATH`
+- The [Red Hat CoreOS][rhcos] image in Glance
+  - This document will use the `rhcos` image name
+- A subnet range for the Nova servers / OpenShift Nodes
+  - This range must not conflict with your existing network
+  - This document will use `192.0.2.0/24`
+- A cluster name you will want to use
+  - This document will use `openshift`
+- A base domain
+  - This document will use `example.com`
+- OpenShift Pull Secret
+- DNS infrastructure you can configure (you'll need to add two records there)
```

Comment:
Do you think we can already be more explicit here?

Maybe something like
```
A DNS zone (or access to `/etc/hosts`) resolvable by the installer machine, for assigning a name to the API external load balancer.
```

or a link to `#create-public-dns-records`, to let the reader figure out what we mean?